command/agent/config/config_test.go

```diff
@@ -11,6 +11,15 @@ import (
 	"github.com/hashicorp/vault/sdk/helper/pointerutil"
 )
 
+var DefaultCustomHeaders = map[string]string{
```

Comment:
I hadn't paid attention to the changes in this file before. It looks like you're modifying Agent's listener config as well.  That may make sense, but it's not clear to me that it was included in the scope, and if it is included, I suspect there need to be additional changes and tests.

Reply:
Well, Agent is its own mode, like `vault agent` vs `vault server`, and has its own listener config.  I guess it's using the same listener parsing code, based on what you're saying.  So we need to decide whether to maintain that and thereby include Agent within the scope of this change, or, more likely, decouple them so that Agent's listener config parsing doesn't have expectations re custom response headers.

I do think it makes sense to fix Agent in the same way, but I'm not sure we want to do that for 1.9 (since we weren't required to) and even if we do, I'd prefer it were its own PR so that we can get this one merged sooner.

Reply:
I am going to decouple it then for now, to do that I figure I could simply add the following in line 177 of command/agent/config/config.go, what would you think? 

`// Pruning custom headers for Agent for now
  for _, ln := range sharedConfig.Listeners {
		ln.CustomResponseHeaders = nil
}
`
well, I will fix the current tests I modified as well. There is a Prune() as well in which I will do the same step as above.